Documentation/hacking.md

```diff
@@ -96,136 +96,49 @@ Follow the instructions on [Update coreos flavor stage1](devel/update-coreos-sta
-(Using [gvm](https://github.com/moovweb/gvm) or other such tomfoolery to manage `GOPATH`s is an exercise left for the reader.)
-
-### Restoring the current state of dependencies
-
-Now that we have a functional `GOPATH`, use `godep` to restore the full set of vendored dependencies to their correct versions.
-(What this command does is essentially just loop over the set of dependencies codified in `Godeps/Godeps.json`, using `go get` to retrieve and then `git checkout` (or equivalent) to set each to their correct revision.)
-
-```
-$ godep restore # might take a while if it's the first time...
-```
-
-At this stage, your path forks, depending on what exactly you want to do: add, update or remove a dependency, or add, update or remove an application.
-But in _all six cases_, the procedure finishes with the [same save command](#saving-the-set-of-dependencies).
+Ensure you have the **latest version** of `glide` version `>= 0.11` and `glide-vc` available in your `PATH`.
 
 #### Add a new dependency
 
-In this case you'll first need to retrieve the dependency you're working with into `GOPATH`.
-As a simple example, assuming we're adding `github.com/fizz/buzz/tazz`:
-
+Use the glide tool to add a new dependency. In order to add a dependency to a package i.e. `github.com/fizz/buzz` for versions `>=1.2.3` and `<2.0.0`, execute:
 ```
-$ go get -d github.com/fizz/buzz
+$ glide get -u github.com/fizz/buzz#^1.2.3
+$ ./scripts/update-glide.sh
 ```
 
-##### If it is a library
+See the glide documentation about [Versions and Ranges](https://github.com/Masterminds/glide/blob/master/docs/versions.md).
 
-Add the new dependency into `godep`'s purview by simply importing the standard package name in one of your sources:
+*Note*: Do *not* use `go get` and `glide update` to add new dependencies. It will cause both `glide.lock` and `glide.yaml` files to diverge.
 
-```
-$ vim $GOPATH/src/github.com/coreos/rkt/some/file.go
-...
-import "github.com/fizz/buzz/tazz"
-...
-```
-
-Now, GOTO [saving](#saving-the-set-of-dependencies)
-
-##### If it is an application
-
-Add the new application to the `vendoredApps` file in the root of the repository:
-
-```
-$ vim vendoredApps
-...
-github.com/fizz/buzz/tazz
-...
-```
-
-Now, GOTO [saving](#saving-the-set-of-dependencies)
-
-#### Update an existing dependency
+#### Update existing dependencies
 
-The steps here are the same for both libraries and applications.
-In this case, assuming we're updating `github.com/foo/bar`:
+Once in a while new versions of dependencies are available. Entries in the `glide.yaml` file specify the target version ranges. If the version range does not suffice the target version, edit the `glade.yaml` directly, or leave it untouched if one wants to simply do a version bump of all dependencies.
 
+Execute the glide update script:
```

Comment:
I'm not sure about the above, wording seems strange and I think several key points are missing. What about the following?
```
To update a vendored dependency to a newer version, first update its target version directly in `glade.yaml`. Glide update script will then take care of pulling all dependencies and refreshing any updated ones, according to version constraints specified in YAML manifest.

Once done editing glide.yaml, execute the glide update script:
```